JASP-Engine/JASP/R/selectionmodels.R

```diff
@@ -939,36 +989,137 @@ jaspResults[["plot_estimates"]] <- plot_estimates
+  
+  if (!class(fit) %in% c("simpleError","error")) {
+    
+    if (fit[["fe"]]) {
+      posMean <- 1    
+    } else {
+      posMean <- 2
+    }
+    
+    fit_old <- fit
+    
+    
+    fit[["unadj_est"]][posMean, 1] <- .smInvTransform(fit_old[["unadj_est"]][posMean, 1], options[["muTransform"]])
+    fit[["ci.lb_adj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.lb_adj"]][posMean, 1], options[["muTransform"]])
+    fit[["ci.ub_adj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.ub_adj"]][posMean, 1], options[["muTransform"]])
+    
+    fit[["adj_est"]][posMean, 1]     <- .smInvTransform(fit_old[["adj_est"]][posMean, 1],     options[["muTransform"]])
+    fit[["ci.lb_unadj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.lb_unadj"]][posMean, 1], options[["muTransform"]])
+    fit[["ci.ub_unadj"]][posMean, 1] <- .smInvTransform(fit_old[["ci.ub_unadj"]][posMean, 1], options[["muTransform"]])
+    
+    fit[["output_unadj"]][["par"]][posMean] <- .smInvTransform(fit_old[["output_unadj"]][["par"]][posMean], options[["muTransform"]])
+    fit[["output_adj"]][["par"]][posMean]   <- .smInvTransform(fit_old[["output_adj"]][["par"]][posMean],   options[["muTransform"]])
+    
+    fit[["estimates_transformed"]] <- options[["muTransform"]]
   }
   
   return(fit)
 }
+
+.smGetInputES              <- function(dataset, options) {
+  
+  # change the direction
+  ES <- dataset[, .v(options[["inputES"]])] * ifelse(options[["effectDirection"]] == "negative", -1, 1)
+  
+  # do a scale transform if neccessary
+  if (options[["measures"]] == "general") {
+    return(ES)
+  } else if (options[["measures"]] == "correlation") {
+    return(.smTransform(ES, transformation = options[["muTransform"]], what = "ES"))
+  }
+}
+
+.smGetInputVar             <- function(dataset, options) {
+  
+  # change the direction
+  ES <- .smGetInputES(dataset, options)
+  ES <- ES * ifelse(options[["effectDirection"]] == "negative", -1, 1)
+  
+  # do a scale transform if neccessary
+  if (options[["measures"]] == "general") {
+    return(dataset[, .v(options[["inputSE"]])]^2)
+  } else if (options[["measures"]] == "correlation") {
+    return(.smTransform(ES, dataset[, .v(options[["inputN"]])], transformation = options[["muTransform"]], what = "VAR"))
+  }
+}
+
+.smGetInputPVal            <- function(dataset, options) {
+  # weightfunc uses one-sided p-values as input!
+  if (options[["inputPVal"]] == "") {
+    pVal <- pnorm(
+      .smGetInputES(dataset, options) / sqrt(.smGetInputVar(dataset, options)),
+      lower.tail = FALSE) 
+  } else {
+    pVal <- dataset[, .v(options[["inputPVal"]])]
+  }
+  return(pVal)
+}
+
+.smTransform               <- function(ES, N = NULL, transformation, what) {
+  if (what == "ES") {
+    return(switch(
+      transformation,
+      "cohensD"  = psych::r2d(ES),
+      "fishersZ" = psych::fisherz(ES)
+    ))
+  } else if (what == "VAR") {
+    if (is.null(N))stop("The effect size needs to be specified for the VAR transformation.")
```

Comment:
I'm a bit confused about this. What is the VAR transformation?

Reply:
it computes variances from the effect sizes. I wanted to bundle both of them in one function

Reply:
I think that "No effect sizes specified to compute the studies' sample variances" might convey that message. I'm guessing variance refers to the sample variance of an individual study, but it might be something else.

Reply:
it does refer to the sampling variances. Changed